iceoryx_binding_c/test/moduletests/test_types_storage_size.hpp

```diff
@@ -34,6 +34,76 @@ extern "C" {
 
 #include <iostream>
 
+inline void verifyStorageSizeCalculationForListener()
+{
+    using namespace ::testing;
+    using namespace ::iox::popo;
+
+    EXPECT_THAT(sizeof(ListenerImpl<1>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(1)));
+    EXPECT_THAT(sizeof(ListenerImpl<2>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(2)));
+    EXPECT_THAT(sizeof(ListenerImpl<3>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(3)));
+    EXPECT_THAT(sizeof(ListenerImpl<4>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(4)));
+    EXPECT_THAT(sizeof(ListenerImpl<5>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(5)));
+    EXPECT_THAT(sizeof(ListenerImpl<6>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(6)));
+    EXPECT_THAT(sizeof(ListenerImpl<7>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(7)));
+    EXPECT_THAT(sizeof(ListenerImpl<8>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(8)));
+
+    EXPECT_THAT(sizeof(ListenerImpl<15>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(15)));
+    EXPECT_THAT(sizeof(ListenerImpl<16>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(16)));
+    EXPECT_THAT(sizeof(ListenerImpl<17>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(17)));
+
+    EXPECT_THAT(sizeof(ListenerImpl<99>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(99)));
+    EXPECT_THAT(sizeof(ListenerImpl<100>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(100)));
+    EXPECT_THAT(sizeof(ListenerImpl<101>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(101)));
+
+    EXPECT_THAT(sizeof(ListenerImpl<255>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(255)));
+    EXPECT_THAT(sizeof(ListenerImpl<256>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(256)));
+    EXPECT_THAT(sizeof(ListenerImpl<257>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(257)));
+
+    EXPECT_THAT(sizeof(ListenerImpl<999>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(999)));
+    EXPECT_THAT(sizeof(ListenerImpl<1000>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(1000)));
+    EXPECT_THAT(sizeof(ListenerImpl<1001>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(1001)));
+
+    EXPECT_THAT(sizeof(ListenerImpl<2047>), Eq(CALCULATE_STORAGE_SIZE_FOR_LISTENER(2047)));
```

Comment:
Is the user able to set these template argument for the ListenerImpl? If not, it is mainly important that it works for the configured value and no number of finite checks will be able to tell us whether it will work for all possible values. You assume there is a linear function (and it likely is), but it does not have to be. The compiler may do something special if it gets larger than page size etc.

Reply:
We have checks for the compiled version. It is in `iceoryx_binding_c/test/moduletests/test_types_storage_size.hpp`. This checks are just there to check the formula.